### DISCUSSION TITLE
[v5] Seeding new tables...

### DISCUSSION THREAD

**Discussion by rbartholomay** [adonisjs/core#2206] [Help] [answered]
Hi there,

i love the way to handle my database via lucid. It is a pleasure to write migrations with create, alter tables.
When a application becomes an update it is possible to have new migration via the batches. All fine..

But how can I handle this for a seeder?  For example:

1. my Application is finished, I deploy it with migration and initial seeding
2. Now my application becomes an update with new tables...

How must I handle the seeder to seed only the new tables?

Bye, Dumbledore

**Comment by McSneaky** [comment]
See those tables in migration 😋 

I'm using quite some migrations that just do something or insert some data 🙂 

For example:
```ts
// populate_industries.ts migration

import BaseSchema from '@ioc:Adonis/Lucid/Schema'
import Industry from 'App/Models/Industry'

export default class PopulateIndustries extends BaseSchema {
  public async up () {
    await Industry.createMany([
      { industry: 'Agriculture' },
      { industry: 'Banking, Finance, Insurance' },
      { industry: 'Construction and Real Estate' },
      { industry: 'Consumer, Retail, Fashion' },
      { industry: 'Education' },
      { industry: 'Energy and Utilitis' },
      { industry: 'Food and Beverage' },
      { industry: 'Gambling & Casinos' },
      { industry: 'Government and Public Sector' },
      { industry: 'Healthcare and Medical' },
      { industry: 'Human Resources' },
      { industry: 'Manufacturing' },
      { industry: 'Marketing, Advertising, PR' },
      { industry: 'Non-profit' },
      { industry: 'Outsourcing & Consulting' },
      { industry: 'Sports, Recreation, Wellness' },
      { industry: 'Technology' },
      { industry: 'Transport, logistics & automotive' },
      { industry: 'Travel, Tourism' },
      { industry: 'Video Games' },
      { industry: 'Other' },
    ])
  }

  public async down () {
  }
}
```

**Comment by thetutlage** [comment, marked as answer]
There are couple of ways to achieve it. 

- You can make your operations inside the seeders idempotent. There is a whole section on it inside the docs https://preview.adonisjs.com/guides/database/seeds#idempotent-operations
- Another is create a manual workflow for yourself. This is how it may look like
	- Create a main seeder file `database/seeders/Index.ts`. You can name it anything you want.
	- Inside this file you can import the other seeders. For example:
		```ts
		import BaseSeeder from '@ioc:Adonis/Lucid/Seeder'
		import OtherSeeder from 'Database/seeders/Other'

		export default class IndexSeeder extends BaseSeeder {
		  public async run() {
		    await new OtherSeeder(this.client).run()
		  }
		}
		```
	- Now only run this file from the command line.
		```
		node ace db:seed --files=./database/seeders/Index
		```

	During the next release just tweak the Index seeder file to not run the old seeders

**Reply by jhosepharaujo** [reply]
I tried to do as you instructed but the following error appeared
```
node ace db:seed --files=./database/seeders/Index
error     database/seeders/Index
Invalid file path. Pass relative path from the application root
```

The file Index.ts exists in seerders folder.

What's happing?

**Reply by yariksav** [reply]
```
node ace db:seed --files=database/seeders/Index
```
or
```
node ace db:seed -f database/seeders/Index
```

**Comment by rbartholomay** [comment]
The solution with the Index.ts is nice!

thanks!
